### ISSUE TITLE
Filter by file content

### ISSUE THREAD

**Issue by hut** [hut/ranger#435]
A feature to display only files whose content matches a certain regex would be rly nice. Slow, yes, but really useful sometimes.

**Comment by lamarpavel** [comment]
Take a look at https://github.com/ggreer/the_silver_searcher, it might be worth a try to delegate this to a highly optimized tool with minimal code addition to ranger.

**Comment by amerlyq** [comment]
I have currently `ag` binded to commands set, allowing search current dir / selected files / filtered files / in flattened view. 
https://github.com/amerlyq/airy/blob/master/cfg/term/ranger/commands.py
It would be really nice to populate results from similar commands into current tab.

**Comment by randynobx** [comment]
@amerlyq Could you repost that commands set? The link is no longer active.

Ever better, could you add it to the commands page in the wiki?

**Comment by amerlyq** [comment]
Yup, it would be better to use direct link to revision at that time, instead of link to head. Nevermind, you found it. 
I don't want to post that to wiki, because currently `:ag` command is crude and full of crotches.
Some of them are result of ranger arch, some are artifacts from my system, some from usage limitation itself.
To integrate filtering smoothly I would need to rewrite at least fifth of ranger codebase to provide API to support displaying of any type of list/tree infromation. This deathwish is too much for me, sorry ;)

**Comment by randynobx** [comment]
I'll probably play around with it and see what I can do with it. Its really nice to have that searching feature in ranger, even if its not fully integrated.

**Comment by amerlyq** [comment]
Be aware, that my command works only in single direction: you can choose by ranger selection what and where to search, but can't load results back into ranger due to ranger arch limitations. Directory pane currectly can't display choosen list of file paths at all. Experimental crotch by hiding all unnecessary files beside found by `:ag` is too cumbersome to actually use -- therefore I do not advice you digging that way, if you aren't liking suffering.

**Comment by suraj-2306** [comment]
probably using `travel` should help, it is mapped to  `scout -aefklst`

**Comment by toonn** [comment]
I don't think travel filters by content. IIRC it filters by filename?

**Comment by suraj-2306** [comment]
> I don't think travel filters by content. IIRC it filters by filename?

Oh I'm sorry. I'll delete my comment

**Comment by toonn** [comment]
That wasn't necessary just makes the thread harder to understand >.<
No worries though.
